// @awx-devx/mcp/dist/server.js
#!/usr/bin/env node
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const node_sdk_1 = require("@airwallex/node-sdk");
const fastmcp_1 = require("fastmcp");
const AirTrackerClient_1 = require("./services/AirTrackerClient");
const createBillingPrice_1 = require("./tools/createBillingPrice");
const createPaymentLink_1 = require("./tools/createPaymentLink");
const createTransfer_1 = require("./tools/createTransfer");
const getFxQuote_1 = require("./tools/getFxQuote");
const index_1 = require("./tools/index");
const listBeneficiaries_1 = require("./tools/listBeneficiaries");
const listPaymentLinks_1 = require("./tools/listPaymentLinks");
const listTransfers_1 = require("./tools/listTransfers");
const simulateTransferUpdate_1 = require("./tools/simulateTransferUpdate");
const version_1 = require("./utils/version");
const withTelemetry_1 = require("./utils/withTelemetry");
const appVersion = (0, version_1.getVersion)(__dirname);
const awxEnv = "demo";
const airTrackerBaseUrl = "https://o11y-demo.airwallex.com";
const disableTelemetry = process.env.DISABLE_TELEMETRY === "true" ||
    process.env.DISABLE_TELEMETRY === "1";
const server = new fastmcp_1.FastMCP({
    name: "Airwallex",
    version: appVersion,
});
const airTrackerClient = new AirTrackerClient_1.AirTrackerClient(airTrackerBaseUrl, awxEnv, appVersion);
// server.addTool({
//   ...searchDocsToolConfig,
//   execute: withToolTelemetry(
//     searchDocsToolConfig.name,
//     async (args) => executeSearchDocs(args, appVersion),
//     { disableTelemetry, telemetryClient: airTrackerClient },
//   ),
// });
server.addTool({
    ...index_1.retrieveDocsToolConfig,
    execute: (0, withTelemetry_1.withToolTelemetry)(index_1.retrieveDocsToolConfig.name, async (args) => (0, index_1.executeRetrieveDocs)(args, appVersion), { disableTelemetry, telemetryClient: airTrackerClient }),
});
server.addTool({
    ...index_1.readBestPracticesToolConfig,
    execute: (0, withTelemetry_1.withToolTelemetry)(index_1.readBestPracticesToolConfig.name, async () => (0, index_1.executeReadBestPractices)(__dirname), { disableTelemetry, telemetryClient: airTrackerClient }),
});
server.addTool({
    ...index_1.validateApiRequestToolConfig,
    execute: (0, withTelemetry_1.withToolTelemetry)(index_1.validateApiRequestToolConfig.name, async (args) => (0, index_1.validateRequest)(args), { disableTelemetry, telemetryClient: airTrackerClient }),
});
if (process.env.AIRWALLEX_SANDBOX_CLIENT_ID &&
    process.env.AIRWALLEX_SANDBOX_API_KEY) {
    const airwallex = new node_sdk_1.Airwallex({
        apiKey: process.env.AIRWALLEX_SANDBOX_API_KEY,
        clientId: process.env.AIRWALLEX_SANDBOX_CLIENT_ID,
        env: "demo",
    });
    server.addTool({
        ...index_1.listGlobalAccountsToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(index_1.listGlobalAccountsToolConfig.name, async () => (0, index_1.executeListGlobalAccounts)(airwallex), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...index_1.getAccountBalancesToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(index_1.getAccountBalancesToolConfig.name, async () => (0, index_1.executeGetAccountBalances)(airwallex), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...index_1.simulateDepositToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(index_1.simulateDepositToolConfig.name, async (args) => (0, index_1.executeSimulateDeposit)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...index_1.listBillingProductsToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(index_1.listBillingProductsToolConfig.name, async () => (0, index_1.executeListBillingProducts)(airwallex), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...index_1.listBillingPricesToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(index_1.listBillingPricesToolConfig.name, async (args) => (0, index_1.executeListBillingPrices)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...index_1.createBillingProductToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(index_1.createBillingProductToolConfig.name, async (args) => (0, index_1.executeCreateBillingProduct)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...createBillingPrice_1.createBillingPriceToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(createBillingPrice_1.createBillingPriceToolConfig.name, async (args) => (0, createBillingPrice_1.executeCreateBillingPrice)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...index_1.createBillingCheckoutToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(index_1.createBillingCheckoutToolConfig.name, async (args) => (0, index_1.executeCreateBillingCheckout)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...getFxQuote_1.getFxQuoteToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(getFxQuote_1.getFxQuoteToolConfig.name, async (args) => (0, getFxQuote_1.executeGetFxQuote)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...listTransfers_1.listTransfersToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(listTransfers_1.listTransfersToolConfig.name, async () => (0, listTransfers_1.executeListTransfers)(airwallex), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...createTransfer_1.createTransferToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(createTransfer_1.createTransferToolConfig.name, async (args) => (0, createTransfer_1.executeCreateTransfer)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...listBeneficiaries_1.listBeneficiariesToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(listBeneficiaries_1.listBeneficiariesToolConfig.name, async () => (0, listBeneficiaries_1.executeListBeneficiaries)(airwallex), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...simulateTransferUpdate_1.simulateTransferUpdateToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(simulateTransferUpdate_1.simulateTransferUpdateToolConfig.name, async (args) => (0, simulateTransferUpdate_1.executeSimulateTransferUpdate)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...createPaymentLink_1.createPaymentLinkToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(createPaymentLink_1.createPaymentLinkToolConfig.name, async (args) => (0, createPaymentLink_1.executeCreatePaymentLink)(airwallex, args), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
    server.addTool({
        ...listPaymentLinks_1.listPaymentLinksToolConfig,
        execute: (0, withTelemetry_1.withToolTelemetry)(listPaymentLinks_1.listPaymentLinksToolConfig.name, async () => (0, listPaymentLinks_1.executeListPaymentLinks)(airwallex), { disableTelemetry, telemetryClient: airTrackerClient }),
    });
}
server.start({
    transportType: "stdio",
});
