// @awx-devx/mcp/dist/services/AirTrackerClient.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AirTrackerClient = void 0;
const os_1 = __importDefault(require("os"));
const uuid_1 = require("uuid");
const device_1 = require("../utils/device");
// FIXME: We could consider including a stable machine ID too.
const DEFAULT_COMMON_DATA = {
    accountId: "unknown",
    apiVersion: "0.0.0",
    appName: "awx-dev-mcp-server-local",
    appVersion: "0.0.0",
    deviceId: (0, device_1.extractDeviceId)(),
    env: "unknown",
    networkType: "unknown",
    platform: os_1.default.platform(),
    sessionId: (0, uuid_1.v4)(),
};
class AirTrackerClient {
    baseUrl;
    appVersion;
    commonData;
    constructor(baseUrl, awxEnv, appVersion) {
        this.baseUrl = baseUrl;
        this.appVersion = appVersion;
        this.commonData = {
            ...DEFAULT_COMMON_DATA,
            appVersion: this.appVersion,
            env: awxEnv,
        };
    }
    logEvent(eventName, properties) {
        const logData = {
            eventName,
            ...properties,
        };
        fetch(`${this.baseUrl}/airtracker/logs`, {
            body: JSON.stringify({
                commonData: this.commonData,
                data: [{ ...logData, severity: "info" }],
            }),
            headers: {
                "Content-Type": "application/json",
                // We do this explicitly because the AirTracker backend has a built-in bot detection check that uses
                // the `isbot` npm library causing false positives. The check is only enforced when the User-Agent header is present.
                // While workaround that, while not ideal, we omit the User-Agent header while making the request.
                "User-Agent": "",
            },
            method: "POST",
            signal: AbortSignal.timeout(2000),
        })
            .then(async (response) => {
            if (!response.ok) {
                console.error(`HTTP error! status: ${response.status}, ${await response.text()}`);
            }
        })
            .catch((error) => {
            console.error(`Error logging event ${eventName}:`, error.message);
        });
    }
}
exports.AirTrackerClient = AirTrackerClient;
