// @awx-devx/mcp/dist/tools/createTransfer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createTransferToolConfig = exports.createTransferSchema = void 0;
exports.executeCreateTransfer = executeCreateTransfer;
const zod_1 = require("zod");
const descriptions_1 = require("../constants/descriptions");
exports.createTransferSchema = zod_1.z.object({
    beneficiary_id: zod_1.z
        .string()
        .min(1)
        .describe("The ID of the beneficiary to send the transfer to."),
    reason: zod_1.z.string().min(1),
    reference: zod_1.z.string().min(1),
    source_currency: zod_1.z
        .string()
        .length(3)
        .describe("The currency of the transfer."),
    transfer_amount: zod_1.z
        .number()
        .positive()
        .describe("The amount of the transfer."),
    transfer_method: zod_1.z
        .enum(["LOCAL", "SWIFT"])
        .describe("The method of the transfer."),
});
async function executeCreateTransfer(airwallex, args) {
    try {
        const response = (await airwallex.post("/api/v1/transfers/create", {
            beneficiary_id: args.beneficiary_id,
            reason: args.reason,
            reference: args.reference,
            request_id: `transfer_${Date.now()}_${Math.random().toString(36).substring(7)}`,
            source_currency: args.source_currency,
            transfer_amount: args.transfer_amount,
            transfer_currency: args.source_currency,
            transfer_method: args.transfer_method,
        }));
        return {
            text: JSON.stringify(response, null, 2),
            type: "text",
        };
    }
    catch (error) {
        const statusCode = error?.status || error?.statusCode || 500;
        const errorMessage = error?.message || "Unknown error occurred";
        throw new Error(`Failed to create transfer (${statusCode}): ${errorMessage}`);
    }
}
exports.createTransferToolConfig = {
    annotations: {
        openWorldHint: false,
        readOnlyHint: false,
        title: "Create transfer",
    },
    description: descriptions_1.TOOL_DESCRIPTIONS.CREATE_TRANSFER,
    name: "sandbox_create_transfer",
    parameters: exports.createTransferSchema,
};
