// @awx-devx/mcp/dist/tools/getAccountBalances.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAccountBalancesToolConfig = exports.getAccountBalancesSchema = void 0;
exports.executeGetAccountBalances = executeGetAccountBalances;
const zod_1 = require("zod");
const descriptions_1 = require("../constants/descriptions");
exports.getAccountBalancesSchema = zod_1.z.object({});
async function executeGetAccountBalances(airwallex) {
    try {
        const response = (await airwallex.get("/api/v1/balances/current"));
        return {
            text: JSON.stringify(response, null, 2),
            type: "text",
        };
    }
    catch (error) {
        const statusCode = error?.status || error?.statusCode || 500;
        const errorMessage = error?.message || "Unknown error occurred";
        throw new Error(`Failed to get account balances (${statusCode}): ${errorMessage}`);
    }
}
exports.getAccountBalancesToolConfig = {
    annotations: {
        openWorldHint: false,
        readOnlyHint: true,
        title: "Get account balances",
    },
    description: descriptions_1.TOOL_DESCRIPTIONS.GET_ACCOUNT_BALANCES,
    name: "sandbox_get_account_balances",
    parameters: exports.getAccountBalancesSchema,
};
