// @awx-devx/mcp/dist/tools/listBillingProducts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listBillingProductsToolConfig = exports.listBillingProductsSchema = void 0;
exports.executeListBillingProducts = executeListBillingProducts;
const zod_1 = require("zod");
const descriptions_1 = require("../constants/descriptions");
exports.listBillingProductsSchema = zod_1.z.object({});
async function executeListBillingProducts(airwallex) {
    try {
        const queryParams = new URLSearchParams();
        queryParams.append("page_size", "100");
        const url = `/api/v1/products${queryParams.toString() ? `?${queryParams.toString()}` : ""}`;
        const response = (await airwallex.get(url));
        return {
            text: JSON.stringify(response, null, 2),
            type: "text",
        };
    }
    catch (error) {
        const statusCode = error?.status || error?.statusCode || 500;
        const errorMessage = error?.message || "Unknown error occurred";
        throw new Error(`Failed to list billing products (${statusCode}): ${errorMessage}`);
    }
}
exports.listBillingProductsToolConfig = {
    annotations: {
        openWorldHint: false,
        readOnlyHint: true,
        title: "List billing products",
    },
    description: descriptions_1.TOOL_DESCRIPTIONS.LIST_BILLING_PRODUCTS,
    name: "sandbox_list_billing_products",
    parameters: exports.listBillingProductsSchema,
};
