// @awx-devx/mcp/dist/tools/listBeneficiaries.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listBeneficiariesToolConfig = exports.listBeneficiariesSchema = void 0;
exports.executeListBeneficiaries = executeListBeneficiaries;
const zod_1 = require("zod");
const descriptions_1 = require("../constants/descriptions");
exports.listBeneficiariesSchema = zod_1.z.object({});
async function executeListBeneficiaries(airwallex) {
    try {
        const queryParams = new URLSearchParams();
        queryParams.append("page_size", "100");
        const url = `/api/v1/beneficiaries${queryParams.toString() ? `?${queryParams.toString()}` : ""}`;
        const response = (await airwallex.get(url));
        return {
            text: JSON.stringify(response, null, 2),
            type: "text",
        };
    }
    catch (error) {
        const statusCode = error?.status || error?.statusCode || 500;
        const errorMessage = error?.message || "Unknown error occurred";
        throw new Error(`Failed to list beneficiaries (${statusCode}): ${errorMessage}`);
    }
}
exports.listBeneficiariesToolConfig = {
    annotations: {
        openWorldHint: false,
        readOnlyHint: true,
        title: "List beneficiaries",
    },
    description: descriptions_1.TOOL_DESCRIPTIONS.LIST_BENEFICIARIES,
    name: "sandbox_list_beneficiaries",
    parameters: exports.listBeneficiariesSchema,
};
