// @awx-devx/mcp/dist/tools/listTransfers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listTransfersToolConfig = exports.listTransfersSchema = void 0;
exports.executeListTransfers = executeListTransfers;
const zod_1 = require("zod");
const descriptions_1 = require("../constants/descriptions");
exports.listTransfersSchema = zod_1.z.object({});
async function executeListTransfers(airwallex) {
    try {
        const response = (await airwallex.get("/api/v1/transfers"));
        return {
            text: JSON.stringify(response, null, 2),
            type: "text",
        };
    }
    catch (error) {
        const statusCode = error?.status || error?.statusCode || 500;
        const errorMessage = error?.message || "Unknown error occurred";
        throw new Error(`Failed to list transfers (${statusCode}): ${errorMessage}`);
    }
}
exports.listTransfersToolConfig = {
    annotations: {
        openWorldHint: false,
        readOnlyHint: true,
        title: "List transfers",
    },
    description: descriptions_1.TOOL_DESCRIPTIONS.LIST_TRANSFERS,
    name: "sandbox_list_transfers",
    parameters: exports.listTransfersSchema,
};
