// @awx-devx/mcp/dist/tools/searchDocs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.searchDocsToolConfig = exports.searchDocsSchema = void 0;
exports.executeSearchDocs = executeSearchDocs;
const zod_1 = require("zod");
const config_1 = require("../constants/config");
const descriptions_1 = require("../constants/descriptions");
exports.searchDocsSchema = zod_1.z.object({
    query: zod_1.z
        .string()
        .min(1)
        .describe("The search query for the documentation. Choose a suitable query depending on the source type. IMPORTANT: NEVER include PII, account IDs, credentials, or sensitive data in queries. Use generic technical terms only."),
    source: zod_1.z
        .enum([
        "product-docs",
        "api-docs",
        "js-sdk-docs",
        "ios-sdk-docs",
        "android-sdk-docs",
    ])
        .describe("The documentation source to search. Use 'product-docs' for integration guides, general product enquiries and detailed API response error code descriptions and reasons, 'api-docs' for API-related information such as endpoints, authentication, rate limits and versioning, and the SDK-specific sources for questions related to their respective SDKs."),
});
async function executeSearchDocs(args, version) {
    const baseUrl = config_1.BASE_URLS.docsSearch;
    const response = await fetch(`${baseUrl}/search`, {
        body: JSON.stringify({
            query: args.query,
            source: args.source,
        }),
        headers: {
            "Content-Type": "application/json",
            "User-Agent": `awx-local-mcp-server/${version}`,
        },
        method: "POST",
    });
    if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
    }
    const data = (await response.json());
    const results = data.results.map((entry) => ({
        content: entry.content,
        title: entry.title,
    }));
    return {
        text: JSON.stringify(results, null, 2),
        type: "text",
    };
}
exports.searchDocsToolConfig = {
    annotations: {
        openWorldHint: false,
        readOnlyHint: true,
        title: "Search Airwallex platform documentation",
    },
    description: descriptions_1.TOOL_DESCRIPTIONS.SEARCH_DOCS,
    name: "search_public_docs",
    parameters: exports.searchDocsSchema,
};
